packages/core/contracts/financial-templates/perpetual-multiparty/ConfigStore.sol

```diff
@@ -0,0 +1,148 @@
+     * @return ConfigSettings config settings that calling financial contract should view as "live".
+     */
+    function getCurrentConfig() external view override nonReentrantView() returns (ConfigSettings memory) {
+        if (_pendingProposalPassed()) {
+            return pendingConfig;
+        } else {
+            return currentConfig;
+        }
+    }
+
+    /**
+     * @notice Propose new configuration settings. New settings go into effect
+     * after a liveness period passes.
+     * @dev Callable only by owner. Calling this while there is already a pending proposal
+     * will overwrite the pending proposal.
+     */
+    function proposeNewConfig(ConfigSettings memory newConfig) external onlyOwner() nonReentrant() updateConfig() {
+        // Alternatively, we can revert if there is already a pending proposal.
+        // require(pendingPassedTimestamp == 0, "Pending proposal exists");
+
+        _validateConfig(newConfig);
+
+        // Warning: This overwrites a pending proposal!
+        pendingConfig = newConfig;
+
+        // Use current config's liveness period to timelock this proposal.
+        pendingPassedTimestamp = getCurrentTime() + currentConfig.updateLiveness;
+
+        emit ProposedNewConfigSettings(
+            msg.sender,
+            newConfig.rewardRatePerSecond.rawValue,
+            newConfig.proposerBondPct.rawValue,
+            newConfig.updateLiveness,
+            pendingPassedTimestamp
+        );
+    }
+
+    function publishPendingConfig() external nonReentrant() updateConfig() {}
+
+    /****************************************
+     *         INTERNAL FUNCTIONS           *
+     ****************************************/
+
+    // Check if pending proposal can overwrite the current config.
+    function _updateConfig() internal {
+        // If liveness has passed, publish new reward rate.
+        if (_pendingProposalPassed()) {
+            currentConfig = pendingConfig;
+
+            _deletePendingConfig();
+
+            emit ChangedNewConfigSettings(
+                currentConfig.rewardRatePerSecond.rawValue,
+                currentConfig.proposerBondPct.rawValue,
+                currentConfig.updateLiveness
+            );
+        }
+    }
+
+    function _deletePendingConfig() internal {
+        delete pendingConfig;
+        pendingPassedTimestamp = 0;
+    }
+
+    function _pendingProposalPassed() internal view returns (bool) {
+        return (pendingPassedTimestamp != 0 && pendingPassedTimestamp <= getCurrentTime());
+    }
+
+    // Use this method to constrain values with which you can set ConfigSettings.
+    function _validateConfig(ConfigSettings memory config) internal pure {
+        // Make sure updateLiveness is not too long, otherwise contract can might not be able to fix itself
+        // before a vulnerability drains its collateral.
+        require(config.updateLiveness <= 604800, "Invalid paramUpdateLiveness");
```

Comment:
nothing else we need to verify?

Reply:
Should we also verify that the reward rate and proposer bond are < 100%?

Reply:
We could prob set upper limits for `rewardRate` and `proposerBond` but its a bit hard to determine at which level these become impractical.

Reply:
Yeah, agreed, we might want reward rate to be even lower. Do you have a good idea of where that threshold should be? Maybe we should come up with an annualized rate that we think is the max.. 100% per year maybe? Thoughts?

I think 100% makes sense for the bond. I could imagine someone wanting to put it over 1% but no real use case for 100%.

Reply:
So the reward % is a bit of a "feel" paramater and we should fully expect to tune it as the contract is live in order to balance the responsiveness of proposers to actual changes in the funding rate. I think 100% a year is a reasonable limit for now (and we can adjust in subsequent perpetuals). 

Regarding the proposer bond, @hlambur and I did some back of the envelope math in [this thread](https://github.com/UMAprotocol/protocol/issues/2039#issuecomment-719734383) and concluded that a 0.04% bond would be equivalent to assuming that the reported funding rate is off from the true funding rate by 200%/year. So I think we should set a limit here << 0.1% per bond.